src/NUnitFramework/tests/Constraints/ThrowsExceptionConstraintTests.cs

```diff
@@ -65,19 +66,39 @@ public void SucceedsWithNonVoidReturningFunction()
         };
 
 #if ASYNC
+        private static async Task YieldAsync()
+        {
+#if NET40
+            await TaskEx.Yield();
+#else
+            await Task.Yield();
+#endif
+        }
+
+
         [Test]
         public static void CatchesAsyncException()
         {
             Assert.That(async () =>
             {
-#if NET40
-                await TaskEx.Yield();
-#else
-                await Task.Yield();
-#endif
+                await YieldAsync();
                 throw new Exception();
             }, Throws.Exception);
         }
+        
+        [Test]
+        public static void CatchesAsyncOfTException()
```

Comment:
`Async` here refers to the way in which the exception is thrown (asynchronously, not synchronously), so we wouldn't treat it as a type and say `AsyncOfT`. Maybe `CatchesAsynchronousTaskOfTException`, since `CatchesSynchronousTaskOfTException` is also a test we should have at some point?

Reply:
To contrast synchronously and asynchronously thrown exceptions:

```c#
async Task ThrowsExceptionAsynchronously() => throw new Exception();
var task = ThrowsExceptionAsynchronously(); // No exception thrown on this line
await task; // Exception thrown on this line

Task AlsoThrowsExceptionAsynchronously() => Task.FromException(new Exception);
var task = AlsoThrowsExceptionAsynchronously(); // No exception thrown on this line
await task; // Exception thrown on this line

Task ThrowsExceptionSynchronously() => throw new Exception();
var task = ThrowsExceptionSynchronously(); // Exception thrown on this line
```

Reply:
Thanks for the suggestions. I've renamed it to `CatchesAsyncTaskOfTException` so it's consistent naming with the existing `CatchesAsyncException` function